### ISSUE TITLE
XRay Receiver does not map links from Xray Segments

### ISSUE THREAD

**Issue by Paulskit** [open-telemetry/opentelemetry-collector-contrib#20350]
### Component(s)

receiver/awsxray

## Description
X-Ray Receiver does not correctly propagate links.

## Steps to Reproduce
Take any X-Ray segment that has links. Example (from BatchGetTraces):
```json
{
  "id": "453dc95c740b9c59",
  "name": "lambda_function",
  "start_time": 1.67993501612e9,
  "trace_id": "1-6421c628-c569cf0ce2427008670d8d45",
  "end_time": 1.679935017219e9,
  "http": { "response": { "status": 200 } },
  "aws": { "request_id": "dcffe126-c03c-558e-aa5e-bdbd087220b5" },
  "origin": "AWS::Lambda",
  "resource_arn": "arn:aws:lambda:us-east-1:123456789012:function:lambda_function",
  "links": [
    {
      "trace_id": "1-6421c610-033542544abf615d316ee2de",
      "id": "2974158e19d0f339",
      "attributes": { "aws.xray.reserved.reference_type": "parent" }
    }
  ]
}
```

This is the output from logging exporter:
```
ResourceSpans #0
Resource SchemaURL: 
Resource attributes:
     -> service.name: Str(lambda_function)
     -> cloud.provider: Str(aws)
     -> aws.xray.resource_arn: Str(arn:aws:lambda:us-east-1:123456789012:function:lambda_function)
ScopeSpans #0
ScopeSpans SchemaURL: 
InstrumentationScope  
Span #0
    Trace ID       : 6421c628c569cf0ce2427008670d8d45
    Parent ID      : 
    ID             : 453dc95c740b9c59
    Name           : lambda_function
    Kind           : Server
    Start time     : 2023-03-27 16:36:56.12 +0000 UTC
    End time       : 2023-03-27 16:36:57.219000064 +0000 UTC
    Status code    : Unset
    Status message : 
Attributes:
     -> http.status_code: Int(200)
     -> aws.request_id: Str(dcffe126-c03c-558e-aa5e-bdbd087220b5)
```

Based on [Logging Exporter Code](https://github.com/open-telemetry/opentelemetry-collector/blob/f6c143b159ea2fdfbec9926d405ace960398f438/exporter/loggingexporter/internal/otlptext/traces.go#L61), if provided, links printed to logs.

## Expected Result
Links from Xray segments are mapped to OTLP Links

## Actual Result

Links from Xray segments are not mapped

### Collector version

v0.72.0

### Environment information

Docker: amazon/aws-otel-collector:latest

### OpenTelemetry Collector configuration

```yaml
extensions:
  health_check:

receivers:
  awsxray:

exporters:
  logging:
    verbosity: 'detailed'

service:
  extensions:
    - health_check
  pipelines:
    traces:
      receivers:
        - awsxray
      exporters:
        - logging
```

**Comment by Paulskit** [comment]
Possibly will be fixed by #20353?

**Comment by willarmiros** [comment]
Thanks for raising this @Paulskit - unfortunately #20353 will not resolve this as it only makes changes on the `awsxrayexporter` side, and this issue is for the receiver. We should keep this open as an enhancement request for the receiver.
